pkg/query-service/app/dashboards/model.go

```diff
@@ -139,7 +139,11 @@ func CreateDashboard(data map[string]interface{}, fm interfaces.FeatureLookup) (
 	dash.CreatedAt = time.Now()
 	dash.UpdatedAt = time.Now()
 	dash.UpdateSlug()
+
 	dash.Uuid = uuid.New().String()
+	if data["uuid"] != nil {
+		dash.Uuid = data["uuid"].(string)
+	}
```

Comment:
yup, that makes sense. 

However, whenever a UUID is given, and no dashboard exists, shouldn't a new one be created with the given UUID from the configuration?

My (possibly biased) use case: I want to provision signoz dashboards with a folder of json files as single source of truth for dashboards. 
A uuid is specified for all of these. The db is fresh, so no dashboards are stored in db. 
The process of starting up signoz should be idempotent, i.e. resulting in all of the dashboards being reflected in the database without duplicates after numerous startups.

With the current logic, if a UUID is specified in json configs, and the UUID does not exist in the db, the dashboard is created with a new UUID because of "CreateDashboard" setting up a new one.  This disregards the uuid specified in the config, which means the configured dashboards would be accounted as not existing in a following provision.

I think this should be accounted for by creating dashboards with the configured UUID, but I might be thinking about it the wrong way or misunderstanding the intent of how provisioning is supposed to work.

Reply:
Your point is fair. We can update the `CreateDashboard` not to create a new ID if it already exists.

Reply:
I've updated the logic now and (regrettably) force pushed to amend commit to hopefully pass commitlint. 
Been testing locally and everything seems to work the way I've intended it in my previous description.

Give me a heads up if there is any other requirement I am missing for this PR to be acceptable.